### PR TITLE
chat permissions check

### PR DESCRIPTION
**Pull request by FrolYo**

### CODE REVIEW

File: src/dotnet/Chat.Service/ChatAuthors.cs [hunk 1]
```diff
@@ -96,19 +97,33 @@ public virtual async Task<ImmutableArray<Symbol>> ListChatIds(Session session, C
...
     public virtual async Task<Presence> GetAuthorPresence(
-        string chatId, string authorId,
+        Session session,
+        string chatId,
+        string authorId,
         CancellationToken cancellationToken)
     {
+        var chat = await Chats.Get(session, chatId, cancellationToken).ConfigureAwait(false);
+        if (chat == null)
+            throw new UnauthorizedAccessException("User does not have access to this chat");
```
Comment (crui3er) [inline src/dotnet/Chat.Service/ChatAuthors.cs:125]: Use StandardError

File: src/dotnet/Chat.Service/Chats.cs [hunk 2]
```diff
@@ -508,11 +517,33 @@ private async Task JoinChat(Session session, string chatId, CancellationToken ca
...
+        CancellationToken cancellationToken)
+    {
+        await RequirePermissions(session, chatEntry.ChatId, ChatPermissions.Write, cancellationToken).ConfigureAwait(false);
+
+        if (chatEntry.IsRemoved)
+            throw StandardError.NotFound<ChatEntry>();
+
+        var author = await ChatAuthors.Get(session, chatEntry.ChatId, cancellationToken).Require().ConfigureAwait(false);
+        if (chatEntry.AuthorId != author.Id)
+            throw new SecurityException("User can remove only their own messages.");
```
Comment (crui3er) [inline src/dotnet/Chat.Service/Chats.cs:544]: Use StandardError, may Unauthorized will suit here

File: src/dotnet/Chat.Service/Chats.cs [hunk 3]
```diff
@@ -508,11 +517,33 @@ private async Task JoinChat(Session session, string chatId, CancellationToken ca
...
+
+        if (chatEntry.IsStreaming)
+            throw new InvalidOperationException("This chat entry is streaming.");
```
Comment (crui3er) [inline src/dotnet/Chat.Service/Chats.cs:547]: Use StandardError

File: src/dotnet/Chat.UI.Blazor/Components/ChatFooter.razor [hunk 4]
```diff
@@ -45,7 +45,7 @@
         var canPost = permissions.CanWrite();
         var canJoin = !canPost && await Chats.CanJoin(Session, chatId, cancellationToken).ConfigureAwait(false);
         return new () {
-            CanPost = canPost,
+            CanPost = canPost = false,
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/ChatFooter.razor:48]: Why do you set 'canPost' to false?

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/ChatFooter.razor (hunk 4 above)
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatFooter.razor]: removed debug code

File: src/dotnet/Chat.Service/Chats.cs (hunk 2 above)
Comment (FrolYo) [reply src/dotnet/Chat.Service/Chats.cs]: done

File: src/dotnet/Chat.Service/Chats.cs (hunk 3 above)
Comment (FrolYo) [reply src/dotnet/Chat.Service/Chats.cs]: done

File: src/dotnet/Chat.Service/ChatAuthors.cs (hunk 1 above)
Comment (FrolYo) [reply src/dotnet/Chat.Service/ChatAuthors.cs]: done

File: src/dotnet/Chat.Service/ChatAuthors.cs [hunk 5]
```diff
@@ -96,19 +97,33 @@ public virtual async Task<ImmutableArray<Symbol>> ListChatIds(Session session, C
...
-        string chatId, string authorId, bool inherit,
+        Session session,
+        string chatId,
+        string authorId,
+        bool inherit,
         CancellationToken cancellationToken)
     {
+        // Check that user has access to chat
+        var chat = await Chats.Get(session, chatId, cancellationToken).ConfigureAwait(false);
+        if (chat == null)
```
Comment (crui3er) [inline src/dotnet/Chat.Service/ChatAuthors.cs:108]: I think we have the same behavior for GetAuthor and GetAuthorPresence if we can not read the chat.
Extract method that checks if we can read the chat.

Comment (FrolYo) [reply src/dotnet/Chat.Service/ChatAuthors.cs:108]: done

**Review by crui3er — APPROVED** [review]

**Pull request merged**